lib/syslog/writer.go

```diff
@@ -244,6 +285,26 @@ func (c bufferedConn) Close() error                { return c.conn.Close() }
 func (c bufferedConn) Flush() error                { return c.buf.Flush() }
 func (c bufferedConn) Write(b []byte) (int, error) { return c.buf.Write(b) }
 
+func writerCacheKey(network string, address string, socksProxy string) string {
+	// We don't currently include the TLS configuration in the cache key. For
+	// now, this is probably OK as the TLS configuration is unlikely to change.
+	return network + address + socksProxy
+}
+
+func cachedDialWriter(network string, address string, config *tls.Config, socksProxy string) (w io.Writer, err error) {
+	writerCache.Lock()
```

Comment:
This global lock on is strange as it causes synchronization of unrelated addresses, the code now cannot open concurrent connections to different addresses, it'll do one after another, if one backend is slow to respond, it also block the other one.

If you don't fix it we should at least leave a comment saying that we know about this, and probably open an issue to track and address at some point.